### ISSUE TITLE
PEP 723 metadata inside a string literal makes a module's own imports unresolvable (regression in 0.0.72)

### ISSUE THREAD

**Issue by ctcjab** [astral-sh/ty#4566]
A shebang followed by a `# /// script` block is detected even when it is inside a string literal. The enclosing file is then resolved as a standalone PEP 723 script, so the project root drops out of the module search path and the file's own first-party imports fail.

This affects any file that embeds an example script in a string — test fixtures, code generators, docs builders.

## Repro

`pkg/lib.py`
```python
def helper() -> int:
    return 1
```

`pkg/user.py`
```python
from pkg.lib import helper

SCRIPT = '''#!/usr/bin/env python3
# /// script
# ///
'''
```

```console
$ ty check .
error[unresolved-import]: Cannot resolve imported module `pkg.lib`
 --> pkg/user.py:1:6
  |
1 | from pkg.lib import helper
  |      ^^^^^^^
info: Searched in the following paths during module resolution:
info:   1. vendored://stdlib (stdlib typeshed stubs vendored by ty)
info:   2. .venv/lib/python3.13/site-packages (site-packages)
```

The project root is absent from the search paths.

## Regression range

0.0.71 passes, 0.0.72 fails. Confirmed still failing on 0.0.73, 0.0.74, 0.0.75, 0.0.76, 0.0.79, 0.0.81, 0.0.82.

## Trigger requires both parts

| content inside the string | result |
|---|---|
| shebang only | clean |
| `# /// script` block only | clean |
| shebang + block | `unresolved-import` |

Any shebang works; it need not be `uv run --script`. An empty block (`# /// script` then `# ///`) is enough.

Independent of whether `__init__.py` is present, and of whether a `pyproject.toml` exists.

## Expected

PEP 723 detection should only consider comments at module level, not text inside string literals.

**Comment by ctcjab** [comment]
**Additional symptom: project configuration stops applying to the affected file.**

Because the file is resolved as a standalone script, `pyproject.toml` no longer governs it — so the diagnostic cannot be suppressed through project config either.

Using the repro above, adding this to `pyproject.toml` has no effect:

```toml
[[tool.ty.overrides]]
include = ["pkg/user.py"]

[tool.ty.overrides.rules]
unresolved-import = "ignore"
```

```console
$ ty check .
Found 1 diagnostic
```

The override mechanism itself is fine. The same override suppresses `invalid-assignment` in an ordinary file, and suppresses a plain `unresolved-import` (`import totally_missing_module`) in a file *without* the embedded script string. It fails only for a file the PEP 723 detection has reclassified. A command-line `-c 'rules.unresolved-import = "ignore"'` still works, which is consistent with project config being skipped for the file.

Practical effect: the only workaround is an inline `# ty: ignore[unresolved-import]` on each import.

**Comment by MichaReiser** [comment]
I agree that this is somewhat unfortunate. The initial script support did use our parser to avoid false positives. But it's fairly costly. We can have another look of what it means performance wise to first to a text based analysis and then do a parser based analysis to avoid false positives, but that can be costly if a project has many scripts and ty's current behavior is in accordance with the PEP (and many tools like uv have the same limitation):

> Note that the specification only requires that top-level comment blocks are recognised as containing metadata. However, parsing Python code is non-trivial, and therefore:
> 
> * Tools MAY choose to do a simple textual scan, rather than a full Python parse
> * As a result of the previous point, the behaviour of scripts that contain data that looks like metadata within another Python construct such as a multi-line string is tool-dependent and should not be relied on.
> * The canonical regular expression provided above is an example of an implementation that does a simple textual scan.
> https://packaging.python.org/en/latest/specifications/inline-script-metadata/#inline-script-metadata

In your case, I'd suggest using

```py
SCRIPT = (
    "#!/usr/bin/env python3\n"
    "# /// script\n"
    "# ///\n"
)
```
